cocotb/decorators.py

```diff
@@ -115,7 +117,7 @@ def __init__(self, inst):
 
         self._task_id = self._id_count
         RunningTask._id_count += 1
-        self.__name__ = "Task %d" % self._task_id
+        self.__name__ = f"{type(self)._name} {self._task_id}"
```

Comment:
If we simply change the name to `type(self).__qualname__` here, we can get rid of the `RunningTest` class. However, that will change the value of `__name__` for tasks from `Task 0` to `RunningTask 0` and `cocotb.coroutine` decorated generators from `Task 0` to `RunningCoroutine 0`.